### PR TITLE
Adding initial docs for websockets security

### PR DESCRIPTION
**Pull request by kinow**
Relates to the Cylc UI Server 0.3 work to verify user authentication with WebSockets: https://github.com/cylc/cylc-uiserver/pull/124, and also to the work to use Tornado's default `check_origin` function: https://github.com/cylc/cylc-uiserver/pull/125

### CODE REVIEW

File: src/security.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+the backend will return an HTTP 403 error and the connection
+will be closed.
+
+If the backend log level is set to ``DEBUG`` a message will be
+emitted in the logs alerting about the unauthorized access.
+
+Same-origin verification
+````````````````````````
+
+TODO: pending Cylc UI Server same-origin verification and docs
```
Comment (kinow) [inline src/security.rst]: This will come after https://github.com/cylc/cylc-uiserver/issues/109

**Comment by kinow** [comment]
image

**Comment by datamel** [comment]
Once this is ready for review, feel free to add me as a review, @kinow

**Comment by kinow** [comment]
> 
> 
> Once this is ready for review, feel free to add me as a review, @kinow I'll add the security detail from the authentication I know about.

Thanks @datamel ! Will ping you once it's ready for review.

File: src/security.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
+Security
+========
+
+TODO: Fill in other parts of this document, with information about
+cylc-flow (jobs, zmq, etc), UI Server, etc.
+
+Web
+~~~
```
Comment (oliver-sanders) [inline src/security.rst:8]: RST lets you use a whole load of characters for different section levels, Sphinx associates these in the order it comes across them.

For ease of editing/refactoring there is a loose convention for Sphinx sectioning.

```
Title
====

Heading
-------

Sub-Heading
^^^^^^^^^^^

Sub-Sub-Heading
""""""""""""""
When these appear it's time to refactor.
```

(note: we've got a documentation writers reference thinggy in the Rose docs - http://metomi.github.io/rose/doc/html/developing/restructuredtext.html#headings)

**Comment by oliver-sanders** [comment]
(note you may need to ignore TB failures on the tutorial suite for now)

File: src/security.rst (hunk 2 above)
Comment (kinow) [reply src/security.rst]: Done in a new commit

File: src/security.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+========
+
+TODO: Fill in other parts of this document, with information about
+cylc-flow (jobs, zmq, etc), UI Server, etc.
+
+Web
+---
+
+TODO: write some introduction about web security in Cylc 8,
+UI Server, OAuth, JupyterHub Authenticators, etc.
```
Comment (hjoliver) [inline src/security.rst:11]: @oliver-sanders and @kinow I guess these TODOs have mostly been addressed by #285. But is there anything else in this old PR that you think we should still add to the docs?

Comment (kinow) [reply src/security.rst:11]: There are a little more detail here I think, but the #285 contains links to JupyterHub docs. The detailed information here could be (or become) outdated if JupyterHub changes how they store cookies (or dropped cookies), and manage websockets.

Might be easier to leave the high-level security information in the new docs, and increase the detail if necessary.

Besides the details, #285 appears to cover everything here, including the TODO's that were planned for later :tada: 

Only thing I'd ask before closing, is whether we should have a page only about security. Searching "security" on GitHub, I found it in the architecture, the summary page, and others. But normally decision-makers, security engineers, some IT operations engineers, etc, may prefer to have a single place to refer about security.

Or maybe we already have it somewhere else that I am not aware @hjoliver ? WDYT?

Comment (hjoliver) [reply src/security.rst:11]: Yeah we should have a security page in the docs, even it mainly just links elsewhere.  I'll post an Issue for that, and close this one.

**Pull request closed without merging**